sdk/trace/src/main/java/io/opentelemetry/sdk/trace/export/BatchSpanProcessorBuilder.java

```diff
@@ -35,6 +36,15 @@ public final class BatchSpanProcessorBuilder {
     this.spanExporter = requireNonNull(spanExporter, "spanExporter");
   }
 
+  /**
+   * Sets whether unsampled spans should be exported. If unset, unsampled spans will not be
+   * exported.
+   */
+  public BatchSpanProcessorBuilder exportUnsampledSpans(boolean exportUnsampledSpans) {
```

Comment:
I was thinking about this.. I think it would be more flexible / future proof to allow the user to set an optional predicate for deciding whether or not a span is exported. Signature might be one of the following:
```
// allow user to make decision using full span
public BatchSpanProcessorBuilder setExportPredicate(Predicate<ReadableSpan>)
// allow user to make decision using full span context
public BatchSpanProcessorBuilder setExportPredicate(Predicate<SpanContext>)
```
Was originally thinking the predicate should accept `SamplingDecision`, but `SamplingDecision` ins't available in `BatchSpanProcessor#onEnd`.

Reply:
> I was also thinking we'd want the processor to always drop unrecorded spans? That'd be indicated by an invalid span context, right?

Yeah, the span never makes it the span processor if its not recorded (see [SdkSpanBuilder](https://github.com/open-telemetry/opentelemetry-java/blob/c8e1e3619e922981922a1034ce07e2a13e99d819/sdk/trace/src/main/java/io/opentelemetry/sdk/trace/SdkSpanBuilder.java#L206-L208)) so no way for the span processor to influence this.